inventory/byo/hosts.origin.example

```diff
@@ -89,6 +89,21 @@ openshift_release=v1.4
 # Skip upgrading Docker during an OpenShift upgrade, leaves the current Docker version alone.
 # docker_upgrade=False
 
+
+# Upgrade Hooks
+#
+# Hooks are available to run custom tasks at various points during a cluster
+# upgrade. Each hook should point to a file with Ansible tasks defined. Suggest using
+# absolute paths, if not the path will be treated as relative to the file where the
+# hook is actually used.
+#
+# Tasks to run before each master is upgraded:
+# openshift_upgrade_pre_master_hook=/usr/share/custom/pre_master.yml
```

Comment:
> I'm thinking we are going to have to store some state on the hosts (using ansible local facts) so that we can track the state of the upgrade. If we go that route, we may need to allow the hooks to have some say in setting their state, to allow for users to tweak hooks part way through a run and have that affect the system state.

:+1: 

> At some point we may have a need for hooks to inject data back into the upgrade. [...]

This makes sense to me, but I'd like to push it out to another chunk of work if possible.